server/executor/poller_executor_test.go

```diff
@@ -457,57 +457,55 @@ func executeAndValidatePollingRequests(t *testing.T, pollerExecutor executor.Pol
 		},
 	}
 
-	for i, value := range expectedValues {
-		request := executor.NewPollingRequest(ctx, test, run, i, pollingprofile.DefaultPollingProfile)
+	// using `pollingprofile.DefaultPollingProfile` and changing the periodic configs directly
+	// causes a race condition because `DefaultPollingProfile.Periodic` is a reference, so it's shared by the copies.
+	// The easiest solution is to create a new polling profile for each test.
+	pp := pollingprofile.PollingProfile{
+		Strategy: pollingprofile.Periodic,
+		Periodic: &pollingprofile.PeriodicPollingConfig{
+			RetryDelay: retryDelay.String(),
+			Timeout:    maxWaitTimeForTrace.String(),
+		},
+	}
 
-		finished, finishReason, anotherRun, err := pollerExecutor.ExecuteRequest(request)
-		run = anotherRun // should store a run to use in another iteration
+	job := executor.NewJob()
```

Comment:
On every call to `NewJob`, we set `Test`, `Run`, and `PollingProfile`. Does it make sense to add them as parameters on `NewJob`?

Reply:
This is not the case for transactions, where it only passes `Transaction` and `TransactionRun`.